game/game.py

```diff
@@ -386,26 +385,50 @@ def set_bullseye(self) -> None:
         self.blue_bullseye = Bullseye(enemy_cp.position)
         self.red_bullseye = Bullseye(player_cp.position)
 
-    def initialize_turn(self) -> None:
+    def initialize_turn(
+        self, for_red: bool = True, for_blue: bool = True, new_turn: bool = True
+    ):
         self.events = []
         self._generate_events()
-
         self.set_bullseye()
 
         # Update statistics
         self.game_stats.update(self)
 
-        self.blue_air_wing.reset()
-        self.red_air_wing.reset()
-        self.aircraft_inventory.reset()
-        for cp in self.theater.controlpoints:
-            self.aircraft_inventory.set_from_control_point(cp)
-
         # Check for win or loss condition
         turn_state = self.check_win_loss()
         if turn_state in (TurnState.LOSS, TurnState.WIN):
             return self.process_win_loss(turn_state)
 
+        # Plan Coalition specific turn
+        if for_red:
+            self.initialize_turn_for(player=False)
+        if for_blue:
+            self.initialize_turn_for(player=True)
+
+        # Plan GroundWar
+        for cp in self.theater.controlpoints:
+            if cp.has_frontline:
+                gplanner = GroundPlanner(cp, self)
+                gplanner.plan_groundwar()
+                self.ground_planners[cp.id] = gplanner
+
+        # Only plan procurement on a new turn
```

Comment:
Why? I think what you want is to _clear_ existing orders (expose a `ControlPoint.refund_all`, which calls `self.pending_unit_deliveries.refund_all`) and then redo the planning. If a base was captured most of the orders will need to be redone, since ground units are now going to the wrong places, aircraft can be delivered in better locations, many planned missions will no longer be valid so the aircraft purchases won't necessarily be needed, etc.

(also should be in `initialize_turn_for`)

Reply:
The problem why i excluded procurement is that the AI will spend all money on the procurement. I was not aware that there is a refund all which will undo the previous procurement order from the current turn. If this can be achieved this would be great. My idead was to touch as less as possible and make the system as stable as possible.

The thing you mention with the Base capture is when using the cheat. so to be complete i say this problem was already existing :D I will have a look into this refund function to see if i can clean the pending procurement from the previous initialize

Reply:
also if i refund the procurement how can i handle a base capture? I would check with is_captured or whatever it was called who currently is in charge of this cp. This would exclude the cheated cp. I have to check the Cheat function more if this also invokes a clear all of this cp 🤔 
Thanks for all the help on this one im getting to know the scary parts of this app now 😆

Reply:
I also struggle with the decision when to refund for player... It would annoy players really badly if the bought units were refunded without even knowing they did so i would go with the approach to only refund for players when they have autobuy enabled:

 ```
for cp in self.theater.controlpoints:
          if (not player and not cp.captured) or (
                player
                and cp.captured
                and self.settings.automate_aircraft_reinforcements
            ):
                cp.pending_unit_deliveries.refund_all(self)
```
Whats your opinion on it @DanAlbert ?

Reply:
> My idea was to touch as less as possible and make the system as stable as possible.

That's my goal as well. I _think_ what I'm recommending is maintaining that, but perhaps there were more pre-existing bugs here than I thought :) Certainly we used to, on capture, refund at least that base's purchase and rerun, but having looked a bit more maybe we weren't covering the other bases before... I think it's an improvement to cover all of the replanning (I expect a cheat capture to be the first thing anyone does on a turn), but we may hear otherwise after it ships.

> I also struggle with the decision when to refund for player... Whats your opinion on it @DanAlbert ?

Agreed. Certainly selling a SAM shouldn't scrap all their plans, which is already covered by not re-init'ing them. Good thinking to check for the automation setting here too. I probably would have forgotten that :) What you have in this block looks good 👍